src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,206 @@
+        /// }
+        /// LocalDeclarationSyntaxNode here is considered as the containing statement of Callee()
+        /// </summary>
+        protected abstract bool ShouldConsideredAsContainingStatement(SyntaxNode syntaxNode);
+        protected abstract TExpression Parenthesize(TExpression node);
+
+        protected AbstractInlineMethodRefactoringProvider(ISyntaxFacts syntaxFacts, ISemanticFactsService semanticFactsService)
+        {
+            _syntaxFacts = syntaxFacts;
+            _semanticFactsService = semanticFactsService;
+        }
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var (document, _, cancellationToken) = context;
+            var calleeMethodInvocationNode = await context.TryGetRelevantNodeAsync<TInvocationNode>().ConfigureAwait(false);
+            if (calleeMethodInvocationNode == null)
+            {
+                return;
+            }
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var calleeMethodSymbol = semanticModel.GetSymbolInfo(calleeMethodInvocationNode, cancellationToken).GetAnySymbol();
+            if (!(calleeMethodSymbol is IMethodSymbol))
+            {
+                return;
+            }
+
+            if (calleeMethodSymbol.DeclaredAccessibility != Accessibility.Private
+                || calleeMethodSymbol.IsConstructor()
+                || calleeMethodSymbol.IsUserDefinedOperator()
+                || calleeMethodSymbol.IsConversion()
+                || calleeMethodSymbol.IsDestructor())
+            {
+                return;
+            }
+
+            var calleeMethodDeclarationSyntaxReferences = calleeMethodSymbol.DeclaringSyntaxReferences;
+            if (calleeMethodDeclarationSyntaxReferences.Length != 1)
+            {
+                return;
+            }
+
+            var calleeMethodDeclarationSyntaxReference = calleeMethodDeclarationSyntaxReferences[0];
+            var calleeMethodDeclarationNode = await calleeMethodDeclarationSyntaxReference.GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+            if (!(calleeMethodDeclarationNode is TMethodDeclarationSyntax))
+            {
+                return;
+            }
+
+            if (!IsSingleStatementOrExpressionMethod((TMethodDeclarationSyntax)calleeMethodDeclarationNode))
+            {
+                return;
+            }
+
+            var statementContainsCallee = GetStatementContainsCallee(calleeMethodInvocationNode);
+            if (statementContainsCallee == null)
+            {
+                return;
+            }
+
+            var invocationOperation = semanticModel.GetOperation(calleeMethodInvocationNode, cancellationToken);
+            if (!(invocationOperation is IInvocationOperation))
+            {
+                return;
+            }
+
+            var codeAction = new CodeAction.DocumentChangeAction(
```

Comment:
you need to sublcass this with a MyCodeAction (See other refactrorings).  this is necessary for telemetry to work.